haystack/components/agents/agent.py

```diff
@@ -296,15 +301,51 @@ def _initialize_fresh_execution(
             tool_invoker_inputs=tool_invoker_inputs,
         )
 
+    def _select_tools(
```

Comment:
It would be great if we could add some unit tests testing some of the error scenarios in this function

Reply:
done